ansible/roles/debops.dovecot/defaults/main.yml

```diff
@@ -281,14 +318,65 @@ dovecot_ssl_cipher_list_better_cypto: 'EDH+CAMELLIA:EDH+aRSA:EECDH+aRSA+AESGCM:E
 # https://www.ncsc.nl/english/current-topics/factsheets/it-security-guidelines-for-transport-layer-security-tls.html
 dovecot_ssl_cipher_list_ncsc_nl: 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-GCM-SHA256'
                                                                    # ]]]
+                                                                   # ]]]
+# Diffie-Hellman parameters [[[
+# -----------------------------
+
+# .. envvar:: dovecot__dhparam [[[
+#
+# Enable or disable support for custom Diffie-Hellman parameters managed by the
+# :ref:`debops.dhparam` Ansible role.
+dovecot__dhparam: '{{ ansible_local.dhparam.enabled
+                      if (ansible_local|d() and ansible_local.dhparam|d() and
+                          ansible_local.dhparam.enabled is defined)
+                      else False }}'
+
+# ]]]
+# .. envvar:: dovecot__dhparam_set [[[
+#
+# Name of the Diffie-Hellman parameter set to use in Dovecot configuration. See
+# :ref:`debops.dhparam` Ansible role for more details.
+dovecot__dhparam_set: 'default'
+
+# ]]]
+# .. envvar:: dovecot__ssl_dh_file [[[
+#
+# Absolute path to Diffie-Hellman parameters file which should be used by Dovecot.
+dovecot__ssl_dh_file: '{{ ansible_local.dhparam[dovecot__dhparam_set]
+                                    if (ansible_local|d() and ansible_local.dhparam|d() and
+                                        ansible_local.dhparam[dovecot__dhparam_set]|d())
+                                    else "" }}'
+
+# ]]]
+# ]]]
 # Dovecot custom configuration [[[
 # --------------------------------
 
 # .. envvar:: dovecot_imap_listeners [[[
 #
 # List of IMAP ``inet_listeners`` which will be enabled. See
 # :ref:`dovecot_imap_listeners` for more details.
-dovecot_imap_listeners: [ 'imap', 'imaps' ]
+dovecot_imap_listeners: '{{ [ "imap", "imaps" ]
+                            if ( dovecot_pki|d(True) )
+                            else [ "imap" ] }}'
+
+                                                                   # ]]]
+# .. envvar:: dovecot_imap_default_config_map [[[
+#
+# Default IMAP configuration properties. See :ref:`dovecot_imap_config_map`
+# for more details.
+dovecot_imap_default_config_map:
+  login-service:
+    inet_listener:
+      imap:
+        # Use only localhost when PKI is true, otherwise listen on all IPs
+        address: '{{ "127.0.0.1, [::0]" if (dovecot_pki|d(True)) else "0.0.0.0, [::]" }}'
```

Comment:
Looks reasonable but breaks compatibility. Just a note.

Reply:
It's a sane secure default, avoid the transmission of clear text passwords over the Interweb, specially when using a default Roundcube setup on another webserver.

Using the same argumentation as for nginx: nginx forwards almost everything from 80 to 443. Therefore we could to something similar with mail.

Reply:
I think that this is backwards. If we have PKI available, IMAP should listen for all connections because IMAP clients exist - we don't have just Roundcube after all. If we don't have PKI available, switching to `localhost` only is reasonable, to be secure.

Your comparsion is right for the wrong reasons - `nginx` provides "public" services which are exepcted to be reachable from anywhere by default, of course over TLS. IMAP in my view is the same - if we provide a mail storage service, offering access via IMAP by default, over TLS, is a reasonable policy, making the Dovecot service "public" as well.

Reply:
So keep :)

Reply:
@reixd What? No, reverse the polarity - put the localhost IP addresses where the `0.0.0.0` and `[::]` are and vice versa, then it will be fine.

Reply:
What I want to achieve is:

```
If (PKI is active) then
  IMAP Listen in v4,v6 localhost addresses;
  IMAPS Listen in all v4,v6 addresses;
Else
  IMAP Listen in all v4,v6 addresses;
  IMAPS is deactivated;
```

Reply:
IMAP over STARTTLS is also a thing. So in this case only enabling or disabling IMAPS makes sense depending on availability of PKI. You should keep IMAP open in both cases.

Reply:
So you want this?
IMAP listen to Internet connections if pki enaed, otherwise keep to localhost?
```
address: '{{ "0.0.0.0, [::]" if (dovecot_pki|d(True)) else "127.0.0.1, [::0]" }}'
```

Reply:
Yes, that sounds reasonable. You could also check the firewall rules and enable/disable them accordingly.